<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Velocyto</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PhD Thesis Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-preprocessing.html">Pre-processing</a>
    </li>
    <li>
      <a href="01B-alevin.html">Alevin</a>
    </li>
    <li>
      <a href="02-quality-control.html">Quality control</a>
    </li>
    <li>
      <a href="03-clustering.html">Clustering</a>
    </li>
    <li>
      <a href="04-marker-genes.html">Marker genes</a>
    </li>
    <li>
      <a href="05-paga.html">PAGA</a>
    </li>
    <li>
      <a href="06-velocyto.html">Velocyto</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lazappi/phd-thesis-analysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Velocyto</h1>

</div>


<p><strong>Last updated:</strong> 2019-02-28</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190110)</code> </summary></p>
<p>The command <code>set.seed(20190110)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="http://github.com/lazappi/phd-thesis-analysis/tree/8948f0899c44cbe951c3461779dc851d8d963522" target="_blank">8948f08</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/cache/
    Ignored:    build-logs/
    Ignored:    data/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/03-clustering/selected_genes.csv.zip
    Ignored:    output/04-marker-genes/de_genes.csv.zip
    Ignored:    packrat/.DS_Store
    Ignored:    packrat/._.DS_Store
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    packrat/src/

Untracked files:
    Untracked:  output/06-velocyto/
    Untracked:  workflowr-build.e2538443
    Untracked:  workflowr-build.e2545370
    Untracked:  workflowr-build.o2538443
    Untracked:  workflowr-build.o2545370

Unstaged changes:
    Modified:   scripts/build.pbs

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="http://github.com/lazappi/phd-thesis-analysis/blob/8948f0899c44cbe951c3461779dc851d8d963522/analysis/06-velocyto.Rmd" target="_blank">8948f08</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-02-28
</td>
<td style="text-align:left;">
Add cell velocity figure
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="http://github.com/lazappi/phd-thesis-analysis/blob/34eb2166bb6137cd33cf4838d1179546d49fd58a/analysis/06-velocyto.Rmd" target="_blank">34eb216</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-02-12
</td>
<td style="text-align:left;">
Add velocyto
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/http://github.com/lazappi/phd-thesis-analysis/34eb2166bb6137cd33cf4838d1179546d49fd58a/docs/06-velocyto.html" target="_blank">34eb216</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-02-12
</td>
<td style="text-align:left;">
Add velocyto
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<pre class="r"><code># scRNA-seq
library(&quot;SingleCellExperiment&quot;)
library(&quot;velocyto.R&quot;)

# Plotting
library(&quot;cowplot&quot;)

# Tidyverse
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>source(here::here(&quot;R/output.R&quot;))</code></pre>
<pre class="r"><code>clust_path &lt;- here::here(&quot;data/processed/03-clustered.Rds&quot;)
loom1_path &lt;- here::here(&quot;data/velocyto/Org1.loom&quot;)
loom2_path &lt;- here::here(&quot;data/velocyto/Org2.loom&quot;)
loom3_path &lt;- here::here(&quot;data/velocyto/Org3.loom&quot;)
loom_paths &lt;- c(loom1_path, loom2_path, loom3_path)</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this document we are to perform cell velocity analysis using velocyto. This approach look at the number of spliced and unspliced reads from each gene and attempts to identify which are being actively transcribed and therefore the direction each cell is differentiating towards.</p>
<pre class="r"><code>if (file.exists(clust_path)) {
    sce &lt;- read_rds(clust_path)
} else {
    stop(&quot;Clustered dataset is missing. &quot;,
         &quot;Please run &#39;03-clustering.Rmd&#39; first.&quot;,
         call. = FALSE)
}

warning(&quot;New clustered dataset loaded, check Loom files are up to date!&quot;)</code></pre>
<p>The first step in using velocyto is to process the aligned BAM files to separately count spliced and unspliced reads. This is done using a command line program and can take a relatively long time so here we start by reading in those results.</p>
<pre class="r"><code>org1 &lt;- read.loom.matrices(loom1_path)
org2 &lt;- read.loom.matrices(loom2_path)
org3 &lt;- read.loom.matrices(loom3_path)

spliced &lt;- cbind(org1$spliced, org2$spliced, org3$spliced)
unspliced &lt;- cbind(org1$unspliced, org2$unspliced, org3$unspliced)

cell_idx &lt;- colData(sce)$Cell
names(cell_idx) &lt;- paste0(
    &quot;Org&quot;, colData(sce)$Sample, &quot;:&quot;, colData(sce)$Barcode, &quot;x&quot;
)

colnames(spliced) &lt;- unname(cell_idx[colnames(spliced)])
spliced &lt;- spliced[intersect(rownames(sce), rownames(spliced)), colnames(sce)]

colnames(unspliced) &lt;- unname(cell_idx[colnames(unspliced)])
unspliced &lt;- unspliced[intersect(rownames(sce), rownames(unspliced)),
                       colnames(sce)]</code></pre>
</div>
<div id="velocyto" class="section level1">
<h1>Velocyto</h1>
<p>We now run velocyto to calclate velocities for each cell. We also project these results onto our previous dimensionality reductions for visualisation.</p>
<pre class="r"><code>spliced &lt;- filter.genes.by.cluster.expression(
    spliced, colData(sce)$Cluster,
    min.max.cluster.average = 0.2
)

unspliced &lt;- filter.genes.by.cluster.expression(
    unspliced, colData(sce)$Cluster,
    min.max.cluster.average = 0.05
)

velocity &lt;- gene.relative.velocity.estimates(
    spliced, unspliced,
    deltaT       = 1,
    kCells       = 30,
    fit.quantile = 0.02,
    n.cores      = 10
)

tSNE_embedding &lt;- show.velocity.on.embedding.cor(
    reducedDim(sce, &quot;SeuratTSNE&quot;), velocity,
    n.cores = 10, show.grid.flow = TRUE, return.details = TRUE
)</code></pre>
<pre class="r"><code>umap_embedding &lt;- show.velocity.on.embedding.cor(
    reducedDim(sce, &quot;SeuratUMAP&quot;), velocity,
    n.cores = 10, show.grid.flow = TRUE, return.details = TRUE
)</code></pre>
<div id="by-cell" class="section level2 tabset">
<h2>By cell</h2>
<p>This plot shows the velocity of individual cells. The direction of each arrow indicates where each cell is headed in this space based on the genes that are being actively transcribed and the length is an indication of rate.</p>
<div id="t-sne" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>tSNE_data &lt;- reducedDim(sce, &quot;SeuratTSNE&quot;) %&gt;%
    as.data.frame() %&gt;%
    mutate(X0 = tSNE_embedding$arrows[, &quot;x0&quot;],
           X1 = tSNE_embedding$arrows[, &quot;x1&quot;],
           Y0 = tSNE_embedding$arrows[, &quot;y0&quot;],
           Y1 = tSNE_embedding$arrows[, &quot;y1&quot;]) %&gt;%
    mutate(X2 = X0 + (X1 - X0) * 4,
           Y2 = Y0 + (Y1 - Y0) * 4) %&gt;%
    mutate(Cluster = colData(sce)$Cluster)

ggplot(tSNE_data) +
    geom_point(aes(x = tSNE_1, y = tSNE_2, colour = Cluster)) +
    geom_segment(aes(x = X0, xend = X2, y = Y0, yend = Y2),
                 arrow = arrow(length = unit(3, &quot;points&quot;), type = &quot;closed&quot;),
                 colour = &quot;grey20&quot;, alpha = 0.8) +
    theme_minimal()</code></pre>
<p><img src="figure/06-velocyto.Rmd/velo-tsne-1.png" width="960" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of velo-tsne-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/lazappi/phd-thesis-analysis/blob/34eb2166bb6137cd33cf4838d1179546d49fd58a/docs/figure/06-velocyto.Rmd/velo-tsne-1.png" target="_blank">34eb216</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-02-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="umap" class="section level3">
<h3>UMAP</h3>
<pre class="r"><code>umap_data &lt;- reducedDim(sce, &quot;SeuratUMAP&quot;) %&gt;%
    as.data.frame() %&gt;%
    setNames(c(&quot;UMAP1&quot;, &quot;UMAP2&quot;)) %&gt;%
    mutate(X0 = umap_embedding$arrows[, &quot;x0&quot;],
           X1 = umap_embedding$arrows[, &quot;x1&quot;],
           Y0 = umap_embedding$arrows[, &quot;y0&quot;],
           Y1 = umap_embedding$arrows[, &quot;y1&quot;]) %&gt;%
    mutate(X2 = X0 + (X1 - X0) * 1,
           Y2 = Y0 + (Y1 - Y0) * 1) %&gt;%
    mutate(Cluster = colData(sce)$Cluster)

ggplot(umap_data) +
    geom_point(aes(x = UMAP1, y = UMAP2, colour = Cluster)) +
    geom_segment(aes(x = X0, xend = X2, y = Y0, yend = Y2),
                 arrow = arrow(length = unit(3, &quot;points&quot;), type = &quot;closed&quot;),
                 colour = &quot;grey20&quot;, alpha = 0.8) +
    theme_minimal()</code></pre>
<p><img src="figure/06-velocyto.Rmd/velo-umap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="vector-field" class="section level2 tabset">
<h2>Vector field</h2>
<p>It can be hard to see all the arrows for individual cells and noise in the data can make similar cells appear to be heading in different directions. Here we summarise the data by building a grid field of vectors that show the average velocity of nearby cells. This gives us a global view of the transcriptional direction of of the dataset.</p>
<div id="t-sne-1" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>tSNE_arrows &lt;- tSNE_embedding$garrows %&gt;%
    as.data.frame() %&gt;%
    mutate(x2 = x0 + (x1 - x0) * 10,
           y2 = y0 + (y1 - y0) * 10)

ggplot(tSNE_data) +
    geom_point(aes(x = tSNE_1, y = tSNE_2, colour = Cluster)) +
    geom_segment(data = tSNE_arrows,
                 aes(x = x0, xend = x2, y = y0, yend = y2),
                 size = 1,
                 arrow = arrow(length = unit(4, &quot;points&quot;), type = &quot;closed&quot;),
                 colour = &quot;grey20&quot;, alpha = 0.8) +
    theme_minimal()</code></pre>
<p><img src="figure/06-velocyto.Rmd/velo-tsne-field-1.png" width="960" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of velo-tsne-field-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/lazappi/phd-thesis-analysis/blob/34eb2166bb6137cd33cf4838d1179546d49fd58a/docs/figure/06-velocyto.Rmd/velo-tsne-field-1.png" target="_blank">34eb216</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-02-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="umap-1" class="section level3">
<h3>UMAP</h3>
<pre class="r"><code>umap_arrows &lt;- umap_embedding$garrows %&gt;%
    as.data.frame() %&gt;%
    mutate(x2 = x0 + (x1 - x0) * 5,
           y2 = y0 + (y1 - y0) * 5)

ggplot(umap_data) +
    geom_point(aes(x = UMAP1, y = UMAP2, colour = Cluster)) +
    geom_segment(data = umap_arrows,
                 aes(x = x0, xend = x2, y = y0, yend = y2),
                 size = 1,
                 arrow = arrow(length = unit(4, &quot;points&quot;), type = &quot;closed&quot;),
                 colour = &quot;grey20&quot;, alpha = 0.8) +
    theme_minimal()</code></pre>
<p><img src="figure/06-velocyto.Rmd/velo-umap-field-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="figures" class="section level1">
<h1>Figures</h1>
<pre class="r"><code>label_data &lt;- umap_data %&gt;%
    group_by(Cluster) %&gt;%
    summarise(UMAP1 = mean(UMAP1),
              UMAP2 = mean(UMAP2))

cell_plot &lt;- ggplot(umap_data) +
    geom_point(aes(x = UMAP1, y = UMAP2, colour = Cluster), alpha = 0.3) +
    geom_segment(aes(x = X0, xend = X2, y = Y0, yend = Y2),
                 arrow = arrow(length = unit(3, &quot;points&quot;), type = &quot;closed&quot;),
                 colour = &quot;grey20&quot;, alpha = 0.3) +
    geom_point(data = label_data, aes(x = UMAP1, y = UMAP2, colour = Cluster),
               shape = 21, size = 6, stroke = 1, fill = &quot;white&quot;) +
    geom_text(data = label_data,
              aes(x = UMAP1, y = UMAP2, colour = Cluster, label = Cluster)) +
    ggtitle(&quot;Individual cell velocity estimates&quot;) +
    theme_minimal() +
    theme(legend.position = &quot;none&quot;)

field_plot &lt;- ggplot(umap_data) +
    geom_point(aes(x = UMAP1, y = UMAP2, colour = Cluster), alpha = 0.3) +
    geom_segment(data = umap_arrows,
                 aes(x = x0, xend = x2, y = y0, yend = y2),
                 arrow = arrow(length = unit(4, &quot;points&quot;), type = &quot;closed&quot;),
                 size = 1, colour = &quot;grey20&quot;, alpha = 0.6) +
    geom_point(data = label_data, aes(x = UMAP1, y = UMAP2, colour = Cluster),
               shape = 21, size = 6, stroke = 1, fill = &quot;white&quot;) +
    geom_text(data = label_data,
              aes(x = UMAP1, y = UMAP2, colour = Cluster, label = Cluster)) +
    ggtitle(&quot;Cell velocity field&quot;) +
    theme_minimal() +
    theme(legend.position = &quot;none&quot;)

fig &lt;- plot_grid(cell_plot, field_plot, nrow = 1, labels = &quot;AUTO&quot;)

ggsave(here::here(&quot;output&quot;, DOCNAME, &quot;cell-velocity.pdf&quot;), fig,
       width = 7, height = 4, scale = 2)
ggsave(here::here(&quot;output&quot;, DOCNAME, &quot;cell-velocity.png&quot;), fig,
       width = 7, height = 4, scale = 2)

fig</code></pre>
<p><img src="figure/06-velocyto.Rmd/figure-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<div id="parameters" class="section level2">
<h2>Parameters</h2>
<p>This table describes parameters used and set in this document.</p>
<pre class="r"><code>params &lt;- list(

)

names(params) &lt;- map_chr(params, magrittr::extract2, &quot;Parameter&quot;)
metadata(sce)$Params[[DOCNAME]] &lt;- params

names(params) &lt;- NULL
params &lt;- jsonlite::toJSON(params, pretty = TRUE)
knitr::kable(jsonlite::fromJSON(params))</code></pre>
<table class="kable_wrapper">
<tbody>
<tr>
<td>
</td>
</tr>
</tbody>
</table>
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<p>This table describes the output files produced by this document. Right click and <em>Save Link As…</em> to download the results.</p>
<pre class="r"><code>dir.create(here::here(&quot;output&quot;, DOCNAME), showWarnings = FALSE)

knitr::kable(data.frame(
    File = c(
        getDownloadLink(&quot;parameters.json&quot;, DOCNAME),
        getDownloadLink(&quot;cell-velocity.png&quot;, DOCNAME),
        getDownloadLink(&quot;cell-velocity.pdf&quot;, DOCNAME)
    ),
    Description = c(
        &quot;Parameters set and used in this analysis&quot;,
        &quot;Cell velocity figure (PNG)&quot;,
        &quot;Cell velocity figure (PDF)&quot;
    )
))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">File</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="http://github.com/lazappi/phd-thesis-analysis/raw/master/output/06-velocyto/parameters.json">parameters.json</a></td>
<td align="left">Parameters set and used in this analysis</td>
</tr>
<tr class="even">
<td align="left"><a href="http://github.com/lazappi/phd-thesis-analysis/raw/master/output/06-velocyto/cell-velocity.png">cell-velocity.png</a></td>
<td align="left">Cell velocity figure (PNG)</td>
</tr>
<tr class="odd">
<td align="left"><a href="http://github.com/lazappi/phd-thesis-analysis/raw/master/output/06-velocyto/cell-velocity.pdf">cell-velocity.pdf</a></td>
<td align="left">Cell velocity figure (PDF)</td>
</tr>
</tbody>
</table>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.0 (2018-04-23)
 os       CentOS release 6.7 (Final)  
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-02-28                  

─ Packages ──────────────────────────────────────────────────────────────
 ! package              * version   date       lib
   assertthat             0.2.0     2017-04-11 [1]
   backports              1.1.3     2018-12-14 [1]
   bindr                  0.1.1     2018-03-13 [1]
   bindrcpp             * 0.2.2     2018-03-29 [1]
   Biobase              * 2.42.0    2018-10-30 [1]
   BiocGenerics         * 0.28.0    2018-10-30 [1]
   BiocParallel         * 1.16.5    2019-01-04 [1]
   bitops                 1.0-6     2013-08-17 [1]
   broom                  0.5.1     2018-12-05 [1]
   callr                  3.1.1     2018-12-21 [1]
   cellranger             1.1.0     2016-07-27 [1]
   cli                    1.0.1     2018-09-25 [1]
 P cluster                2.0.7-1   2018-04-13 [5]
 P codetools              0.2-15    2016-10-05 [5]
   colorspace             1.4-0     2019-01-13 [1]
   cowplot              * 0.9.4     2019-01-08 [1]
   crayon                 1.3.4     2017-09-16 [1]
   DelayedArray         * 0.8.0     2018-10-30 [1]
   desc                   1.2.0     2018-05-01 [1]
   devtools               2.0.1     2018-10-26 [1]
   digest                 0.6.18    2018-10-10 [1]
   dplyr                * 0.7.8     2018-11-10 [1]
   evaluate               0.12      2018-10-09 [1]
   forcats              * 0.3.0     2018-02-19 [1]
   fs                     1.2.6     2018-08-23 [1]
   generics               0.0.2     2018-11-29 [1]
   GenomeInfoDb         * 1.18.1    2018-11-12 [1]
   GenomeInfoDbData       1.2.0     2019-01-15 [1]
   GenomicRanges        * 1.34.0    2018-10-30 [1]
   ggplot2              * 3.1.0     2018-10-25 [1]
   git2r                  0.24.0    2019-01-07 [1]
   glue                   1.3.0     2018-07-17 [1]
   gtable                 0.2.0     2016-02-26 [1]
   haven                  2.0.0     2018-11-22 [1]
   here                   0.1       2017-05-28 [1]
   highr                  0.7       2018-06-09 [1]
   hms                    0.4.2     2018-03-10 [1]
   htmltools              0.3.6     2017-04-28 [1]
   httr                   1.4.0     2018-12-11 [1]
   IRanges              * 2.16.0    2018-10-30 [1]
   jsonlite               1.6       2018-12-07 [1]
   knitr                  1.21      2018-12-10 [1]
 P lattice                0.20-35   2017-03-25 [5]
   lazyeval               0.2.1     2017-10-29 [1]
   lubridate              1.7.4     2018-04-11 [1]
   magrittr               1.5       2014-11-22 [1]
 P MASS                   7.3-50    2018-04-30 [5]
 P Matrix               * 1.2-14    2018-04-09 [5]
   matrixStats          * 0.54.0    2018-07-23 [1]
   memoise                1.1.0     2017-04-21 [1]
 P mgcv                   1.8-24    2018-06-18 [5]
   modelr                 0.1.3     2019-02-05 [1]
   munsell                0.5.0     2018-06-12 [1]
 P nlme                   3.1-137   2018-04-07 [5]
   pcaMethods             1.74.0    2018-10-30 [1]
   pillar                 1.3.1     2018-12-15 [1]
   pkgbuild               1.0.2     2018-10-16 [1]
   pkgconfig              2.0.2     2018-08-16 [1]
   pkgload                1.0.2     2018-10-29 [1]
   plyr                   1.8.4     2016-06-08 [1]
   prettyunits            1.0.2     2015-07-13 [1]
   processx               3.2.1     2018-12-05 [1]
   ps                     1.3.0     2018-12-21 [1]
   purrr                * 0.3.0     2019-01-27 [1]
   R.methodsS3            1.7.1     2016-02-16 [1]
   R.oo                   1.22.0    2018-04-22 [1]
   R.utils                2.7.0     2018-08-27 [1]
   R6                     2.3.0     2018-10-04 [1]
   Rcpp                   1.0.0     2018-11-07 [1]
   RCurl                  1.95-4.11 2018-07-15 [1]
   readr                * 1.3.1     2018-12-21 [1]
   readxl                 1.2.0     2018-12-19 [1]
   remotes                2.0.2     2018-10-30 [1]
   rlang                  0.3.1     2019-01-08 [1]
   rmarkdown              1.11      2018-12-08 [1]
   rprojroot              1.3-2     2018-01-03 [1]
   rstudioapi             0.9.0     2019-01-09 [1]
   rvest                  0.3.2     2016-06-17 [1]
   S4Vectors            * 0.20.1    2018-11-09 [1]
   scales                 1.0.0     2018-08-09 [1]
   sessioninfo            1.1.1     2018-11-05 [1]
   SingleCellExperiment * 1.4.1     2019-01-04 [1]
   stringi                1.2.4     2018-07-20 [1]
   stringr              * 1.3.1     2018-05-10 [1]
   SummarizedExperiment * 1.12.0    2018-10-30 [1]
   testthat               2.0.0     2017-12-13 [5]
   tibble               * 2.0.1     2019-01-12 [1]
   tidyr                * 0.8.2     2018-10-28 [1]
   tidyselect             0.2.5     2018-10-11 [1]
   tidyverse            * 1.2.1     2017-11-14 [1]
   usethis                1.4.0     2018-08-14 [1]
   velocyto.R           * 0.6       2019-02-10 [1]
   whisker                0.3-2     2013-04-28 [1]
   withr                  2.1.2     2018-03-15 [1]
   workflowr              1.1.1     2018-07-06 [1]
   xfun                   0.4       2018-10-23 [1]
   xml2                   1.2.0     2018-01-24 [1]
   XVector                0.22.0    2018-10-30 [1]
   yaml                   2.2.0     2018-07-25 [1]
   zlibbioc               1.28.0    2018-10-30 [1]
 source                                   
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 Bioconductor                             
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 Bioconductor                             
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Github (velocyto-team/velocyto.R@666e1db)
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 CRAN (R 3.5.0)                           
 Bioconductor                             
 CRAN (R 3.5.0)                           
 Bioconductor                             

[1] /group/bioi1/luke/analysis/phd-thesis-analysis/packrat/lib/x86_64-pc-linux-gnu/3.5.0
[2] /group/bioi1/luke/analysis/phd-thesis-analysis/packrat/lib-ext/x86_64-pc-linux-gnu/3.5.0
[3] /group/bioi1/luke/analysis/phd-thesis-analysis/packrat/lib-R/x86_64-pc-linux-gnu/3.5.0
[4] /home/luke.zappia/R/x86_64-pc-linux-gnu-library/3.5
[5] /usr/local/installed/R/3.5.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
