<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Methods</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PhD Thesis Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="getting-started.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-preprocessing.html">Pre-processing</a>
    </li>
    <li>
      <a href="01B-alevin.html">Alevin</a>
    </li>
    <li>
      <a href="02-quality-control.html">Quality control</a>
    </li>
    <li>
      <a href="03-clustering.html">Clustering</a>
    </li>
    <li>
      <a href="04-marker-genes.html">Marker genes</a>
    </li>
    <li>
      <a href="05-paga.html">PAGA</a>
    </li>
    <li>
      <a href="06-velocyto.html">Cell velocity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Reports
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="cellranger_web_summary.html">Cell Ranger</a>
    </li>
  </ul>
</li>
<li>
  <a href="90-methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/lazappi/phd-thesis-analysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Methods</h1>

</div>


<p><strong>Last updated:</strong> 2019-04-03</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190110)</code> </summary></p>
<p>The command <code>set.seed(20190110)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="http://github.com/lazappi/phd-thesis-analysis/tree/5870e6e191cf7f91714b40501fc63645582ed59a" target="_blank">5870e6e</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/cache/
    Ignored:    build-logs/
    Ignored:    data/alevin/
    Ignored:    data/cellranger/
    Ignored:    data/processed/
    Ignored:    data/published/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/03-clustering/selected_genes.csv.zip
    Ignored:    output/04-marker-genes/de_genes.csv.zip
    Ignored:    packrat/.DS_Store
    Ignored:    packrat/._.DS_Store
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    packrat/src/

Untracked files:
    Untracked:  DGEList.Rds
    Untracked:  output/90-methods/package-versions.json
    Untracked:  scripts/build.pbs

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   output/01-preprocessing/droplet-selection.pdf
    Modified:   output/01-preprocessing/parameters.json
    Modified:   output/01-preprocessing/selection-comparison.pdf
    Modified:   output/01B-alevin/alevin-comparison.pdf
    Modified:   output/01B-alevin/parameters.json
    Modified:   output/02-quality-control/qc-thresholds.pdf
    Modified:   output/02-quality-control/qc-validation.pdf
    Modified:   output/03-clustering/cluster-comparison.pdf
    Modified:   output/03-clustering/cluster-validation.pdf
    Modified:   output/04-marker-genes/de-results.pdf

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="http://github.com/lazappi/phd-thesis-analysis/blob/5870e6e191cf7f91714b40501fc63645582ed59a/analysis/90-methods.Rmd" target="_blank">5870e6e</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-04-03
</td>
<td style="text-align:left;">
Adjust figures and fix names
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="http://github.com/lazappi/phd-thesis-analysis/blob/33ac14f1f1aa822355526e1696a4b10e25e79226/analysis/90-methods.Rmd" target="_blank">33ac14f</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-03-20
</td>
<td style="text-align:left;">
Tidy up website
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/http://github.com/lazappi/phd-thesis-analysis/33ac14f1f1aa822355526e1696a4b10e25e79226/docs/90-methods.html" target="_blank">33ac14f</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-03-20
</td>
<td style="text-align:left;">
Tidy up website
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/http://github.com/lazappi/phd-thesis-analysis/2693e979c9a55c66e7d99877e14f641a06a9b9a7/docs/90-methods.html" target="_blank">2693e97</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2019-03-05
</td>
<td style="text-align:left;">
Add methods page
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<pre class="r"><code># Presentation
library(&quot;glue&quot;)
library(&quot;knitr&quot;)

# JSON
library(&quot;jsonlite&quot;)

# Tidyverse
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>dir.create(here::here(&quot;output&quot;, DOCNAME), showWarnings = FALSE)

write_bib(c(&quot;base&quot;, &quot;SingleCellExperiment&quot;, &quot;cowplot&quot;, &quot;UpSetR&quot;, &quot;ggforce&quot;,
            &quot;gridExtra&quot;, &quot;tidyverse&quot;, &quot;dplyr&quot;, &quot;tidyr&quot;, &quot;purrr&quot;,
            &quot;LoomExperiment&quot;, &quot;workflowr&quot;, &quot;knitr&quot;, &quot;rmarkdown&quot;),
          file = here::here(&quot;output&quot;, DOCNAME, &quot;packages.bib&quot;))</code></pre>
<pre class="r"><code>docs &lt;- list.dirs(here::here(&quot;output&quot;), full.names = FALSE)[-1]

params &lt;- map(docs, function(doc) {
    out &lt;- tryCatch(
        {
            table &lt;- suppressWarnings(
                read_json(here::here(glue(&quot;output/{doc}/parameters.json&quot;)),
                          simplifyVector = TRUE)
            )
            p_list &lt;- table$Value %&gt;% setNames(table$Parameter)
        },
        error = function(e) {
            message(glue(&quot;{doc} parameters file not found&quot;))

            return(list())
        }
    )    
    
    return(out)
})

names(params) &lt;- str_remove(docs, &quot;[0-9A-Z]+\\-&quot;)
names(params) &lt;- names(params) %&gt;%
    str_replace(&quot;preprocessing&quot;, &quot;pre&quot;) %&gt;%
    str_replace(&quot;quality-control&quot;, &quot;qc&quot;) %&gt;%
    str_replace(&quot;clustering&quot;, &quot;clust&quot;) %&gt;%
    str_replace(&quot;marker-genes&quot;, &quot;markers&quot;)

versions &lt;- list(
    biomaRt        = packageVersion(&quot;biomaRt&quot;),
    cellranger     = &quot;3.0.1&quot;,
    cellranger_ref = &quot;3.0.0&quot;,
    clustree       = packageVersion(&quot;clustree&quot;),
    cowplot        = packageVersion(&quot;cowplot&quot;),
    dplyr          = packageVersion(&quot;dplyr&quot;),
    DropletUtils   = packageVersion(&quot;DropletUtils&quot;),
    edgeR          = packageVersion(&quot;edgeR&quot;),
    ensembl        = &quot;93&quot;,
    ggforce        = packageVersion(&quot;ggforce&quot;),
    ggplot2        = packageVersion(&quot;ggplot2&quot;),
    gridExtra      = packageVersion(&quot;gridExtra&quot;),
    knitr          = packageVersion(&quot;knitr&quot;),
    LoomExperiment = packageVersion(&quot;LoomExperiment&quot;),
    M3Drop         = packageVersion(&quot;M3Drop&quot;),
    pandoc         = rmarkdown::pandoc_version(),
    purrr          = packageVersion(&quot;purrr&quot;),
    python         = &quot;3.6.8&quot;,
    R              = str_extract(R.version.string, &quot;[0-9\\.]+&quot;),
    rmarkdown      = packageVersion(&quot;rmarkdown&quot;),
    salmon         = &quot;0.12.0&quot;,
    scanpy         = &quot;1.4&quot;,
    scater         = packageVersion(&quot;scater&quot;),
    SCE            = packageVersion(&quot;SingleCellExperiment&quot;),
    scran          = packageVersion(&quot;scran&quot;),
    Seurat         = packageVersion(&quot;Seurat&quot;),
    tidyr          = packageVersion(&quot;tidyr&quot;),
    tidyverse      = packageVersion(&quot;tidyverse&quot;),
    UpSetR         = packageVersion(&quot;UpSetR&quot;),
    velocyto.py    = &quot;0.17.16&quot;,
    velocyto.R     = packageVersion(&quot;velocyto.R&quot;),
    viridis        = packageVersion(&quot;viridis&quot;),
    workflowr      = packageVersion(&quot;workflowr&quot;)
)</code></pre>
<div id="pre-processing" class="section level1">
<h1>Pre-processing</h1>
<p>The Cell Ranger pipeline (v3.0.1) <span class="citation">(Zheng et al. 2017)</span> was used to perform sample demultiplexing, barcode processing and single-cell gene counting. Briefly,samples were demultiplexed to produce a pair of FASTQ files for each sample. Reads containing sequence information were aligned using the reference provided with Cell Ranger (v3.0.0) based on the GRCh38 reference genome and ENSEMBL gene annotation (v93). PCR duplicates were removed by selecting unique combinations of cell barcodes, unique molecular identifiers and gene ids with the final results being a gene expression matrix that was used for further analysis. The three samples were aggregated using Cell Ranger with no normalisation and treated as a single dataset containing 945038 droplets. The R statistical programming language (v3.5.0) <span class="citation">(R Core Team 2018)</span> was used for further analysis. Count data was read into R and used to construct a SingleCellExperiment object (v1.4.1) <span class="citation">(A. Lun and Risso 2019)</span>.</p>
<div id="droplet-selection" class="section level2">
<h2>Droplet selection</h2>
<p>The droplet selection method in this version of Cell Ranger identified 8291 cell containing droplets. This is more than the traditional Cell Ranger approach based on a simple threshold on total counts which identified 7006 cells. The EmptyDrops method <span class="citation">(A. T. L. Lun et al. 2019)</span> available in the DropletUtils package (v1.2.2) was also applied. Droplets with less than 100 total counts were used to construct the ambient RNA profile and an FDR threshold of less than 0.01 was used to select 9858 droplets. The droplets selected by either the Cell Ranger v3 method or EmptyDrops were kept giving a dataset with 10002 cells. Gene annotation information was added from BioMart <span class="citation">(Smedley et al. 2015)</span> using the biomaRt package (v2.38.0) <span class="citation">(Durinck et al. 2005)</span> and cells were assigned cell cycle scores using the cyclone <span class="citation">(Scialdone et al. 2015)</span> function in the scran package (v1.10.2) <span class="citation">(A. T. L. Lun, Bach, and Marioni 2016)</span>.</p>
</div>
<div id="alevin" class="section level2">
<h2>Alevin</h2>
<p>An alternative quantification was performed using the alevin method <span class="citation">(Srivastava et al. 2019)</span> available as part of the salmon software package (v0.12.0) <span class="citation">(Patro et al. 2017)</span>. This produced a dataset with 10423 cells and 37247 genes.</p>
</div>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>The scater package (v1.10.1) <span class="citation">(McCarthy et al. 2017)</span> was used to produce a series of diagnostic quality control plots and select thresholds for filtering. Cells were removed from the dataset if they had <span class="math inline">\(\log_{10}\)</span> total counts less than 2.309 (4 MADs from the median), <span class="math inline">\(\log_{10}\)</span> total features expressed less than 2.3659 (4 MADs) or more than 8.2411 percent of counts (4 MADs) assigned to mitochondrial genes. An automated filtering method method available in scater that detects outliers in a PCA of cells based on technical factors was also performed for comparison but not used for filtering. After filtering doublet scores for each cell were calculated using the simulation-based method available in scran and cells with a score greater than 1.5572 were removed. A minimal filter was used to remove genes that did not have at least 1 count in at least 2 cells. After quality control the dataset had 8059 cells and 22739 genes with a median of 7838 counts per cell and a median of 2506 genes expressed.</p>
</div>
<div id="clustering" class="section level1">
<h1>Clustering</h1>
<div id="gene-selection" class="section level2">
<h2>Gene selection</h2>
<p>Two methods were used to select genes to use for clustering. The default selection method in the Seurat package (v2.3.4) <span class="citation">(Satija et al. 2015)</span> uses thresholds on mean expression (between 0.0125 and 3.5) and dispersion (greater than 1), which selected 2457 genes. The M3Drop (v3.10.3) <span class="citation">(T. S. Andrews and Hemberg 2018)</span> method was also used to test for genes with significantly more zero counts than expected and 1952 genes with an adjusted p-value less than 0.01 were selected. Following comparisions between these two methods the genes identified by M3Drop were used for clustering.</p>
</div>
<div id="graph-based-clustering" class="section level2">
<h2>Graph-based clustering</h2>
<p>Seurat was used to perform graph-based clustering. PCA was performed using the selected genes and the first 15 principal components were used to construct a shared nearest neighbour graph using the overlap between the 30 nearest neighbours of each cell. Louvain modularity optimisation <span class="citation">(Blondel et al. 2008)</span> was used to partition this graph with a resolution parameter between 0 and 1. Clustering tree visualisations <span class="citation">(Zappia and Oshlack 2018)</span> were produced using the clustree package (v0.3.0) showing the expression of previously identified marker genes. By inspecting these trees a resolution of 0.4 which chosen which produced 13 clusters. To compare these cluster to those that had been previously published the Jaccard index was calculated between pairs of clusters from each analysis and visualised as a heatmap.</p>
</div>
</div>
<div id="marker-genes" class="section level1">
<h1>Marker genes</h1>
<p>Marker genes for each cluster were identified using edgeR (v3.24.3) <span class="citation">(Robinson, McCarthy, and Smyth 2010; McCarthy, Chen, and Smyth 2012)</span>. Additional filtering was performed to remove genes with a <span class="math inline">\(\log_{10}\)</span> maximum average counts in any cluster less than -1.3505. The edgeR negative binomial model was then fitted to the dataset using a design matrix that included the detection rate (scaled proportion of genes expressed in each cell). The quasi-likelihood F-test was used to test cells in one cluster against all other cells. Genes were considered significant markers for a cluster if they had an FDR less than 0.05 and a log foldchange greater than 1. Identities were assigned to each cluster by comparing the detected genes to previously published lists of cell type markers.</p>
</div>
<div id="partition-based-graph-abstraction-paga" class="section level1">
<h1>Partition-based graph abstraction (PAGA)</h1>
<p>Partition-based graph abstraction (PAGA) <span class="citation">(Wolf et al. 2019)</span> was performed using the scanpy library (v1.4) <span class="citation">(Wolf, Angerer, and Theis 2018)</span> for the Python programming language (v3.6.8). A shared nearest neighbour graph was built using the same parameters as were used by Seurat and connectivity between the Seurat clusters was calculated using the PAGA algorithm. A ForceAtlas2 <span class="citation">(Jacomy et al. 2014)</span> layout of the cell graph was calculated based on the PAGA cluster graph. The results of the PAGA analysis were read into R for visualisation.</p>
</div>
<div id="cell-velocity" class="section level1">
<h1>Cell velocity</h1>
<p>The velocyto (v0.17.16) <span class="citation">(La Manno et al. 2018)</span> Python program was used to calculate spliced and unspliced count matrices from the aligned reads. The resulting Loom files for each sample were then read into R and merged. The remaining analysis was performed using the velocyto R package (v0.6). Genes were removed from the spliced matrix if they had did not have a mean expression greater than 0.2 in at least one cluster or from the unspliced matrix if they did not have a mean expression greater than 0.05 in at least one cluster. Relative velocity estimates were then calculated for each cell and projected onto reduced dimensional spaces for visualisation.</p>
</div>
<div id="other-packages" class="section level1">
<h1>Other packages</h1>
<p>Visualisations and figures were primarily created using the ggplot2 (v3.1.0) <span class="citation">(Wickham 2010)</span> and cowplot (v0.9.4) <span class="citation">(Wilke 2019)</span> packages using the viridis colour palette (v0.5.1) for continuous data. UpSet plots <span class="citation">(Lex et al. 2014)</span> were produced using the UpSetR package (v1.3.3) <span class="citation">(Gehlenborg 2017)</span> with help from the gridExtra package (v2.3) <span class="citation">(Auguie 2017)</span> and SinaPlots <span class="citation">(Sidiropoulos et al. 2018)</span> using the ggforce package (v0.1.3) <span class="citation">(Pedersen 2018)</span>. Data manipulation was performed using other packages in the tidyverse (v1.2.1) <span class="citation">(Wickham 2017)</span> particularly dplyr (v0.7.8) <span class="citation">(Wickham et al. 2018)</span>, tidyr (v0.7.8) <span class="citation">(Wickham and Henry 2018)</span> and purrr (v0.3.0) <span class="citation">(Henry and Wickham 2019)</span>. Loom files were exported from R using the LoomExperiment package (v1.0.2) <span class="citation">(Morgan and Van Twisk 2019)</span>. The analysis project was managed using the workflowr (v1.1.1) <span class="citation">(Blischak, Carbonetto, and Stephens 2018)</span> package which was also used to produce the publicly available website displaying the analysis code, results and output. Reproducible reports were produced using knitr (v1.21) <span class="citation">(Xie 2014; Xie 2016; Xie 2018)</span> and R Markdown (v1.11) <span class="citation">(Xie, Allaire, and Grolemund 2018; J. Allaire et al. 2018)</span> and converted to HTML using Pandoc (v1.17.2).</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs">
<div id="ref-R-rmarkdown">
<p>Allaire, JJ, Yihui Xie, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, Hadley Wickham, Joe Cheng, Winston Chang, and Richard Iannone. 2018. <em>Rmarkdown: Dynamic Documents for R</em>. <a href="https://CRAN.R-project.org/package=rmarkdown" class="uri">https://CRAN.R-project.org/package=rmarkdown</a>.</p>
</div>
<div id="ref-Andrews2018-xa">
<p>Andrews, T S, and Martin Hemberg. 2018. “M3Drop: Dropout-based feature selection for scRNASeq.” <em>Bioinformatics</em>, December. doi:<a href="https://doi.org/10.1093/bioinformatics/bty1044">10.1093/bioinformatics/bty1044</a>.</p>
</div>
<div id="ref-R-gridExtra">
<p>Auguie, Baptiste. 2017. <em>GridExtra: Miscellaneous Functions for “Grid” Graphics</em>. <a href="https://CRAN.R-project.org/package=gridExtra" class="uri">https://CRAN.R-project.org/package=gridExtra</a>.</p>
</div>
<div id="ref-R-workflowr">
<p>Blischak, John, Peter Carbonetto, and Matthew Stephens. 2018. <em>Workflowr: A Framework for Reproducible and Collaborative Data Science</em>. <a href="https://CRAN.R-project.org/package=workflowr" class="uri">https://CRAN.R-project.org/package=workflowr</a>.</p>
</div>
<div id="ref-Blondel2008-ym">
<p>Blondel, Vincent D, Jean-Loup Guillaume, Renaud Lambiotte, and Etienne Lefebvre. 2008. “Fast unfolding of communities in large networks.” <em>Journal of Statistical Mechanics</em> 2008 (10). IOP Publishing: P10008. doi:<a href="https://doi.org/10.1088/1742-5468/2008/10/P10008">10.1088/1742-5468/2008/10/P10008</a>.</p>
</div>
<div id="ref-Durinck2005-zb">
<p>Durinck, Steffen, Yves Moreau, Arek Kasprzyk, Sean Davis, Bart De Moor, Alvis Brazma, and Wolfgang Huber. 2005. “BioMart and Bioconductor: a powerful link between biological databases and microarray data analysis.” <em>Bioinformatics</em> 21 (16): 3439–40. doi:<a href="https://doi.org/10.1093/bioinformatics/bti525">10.1093/bioinformatics/bti525</a>.</p>
</div>
<div id="ref-R-UpSetR">
<p>Gehlenborg, Nils. 2017. <em>UpSetR: A More Scalable Alternative to Venn and Euler Diagrams for Visualizing Intersecting Sets</em>. <a href="https://CRAN.R-project.org/package=UpSetR" class="uri">https://CRAN.R-project.org/package=UpSetR</a>.</p>
</div>
<div id="ref-R-purrr">
<p>Henry, Lionel, and Hadley Wickham. 2019. <em>Purrr: Functional Programming Tools</em>. <a href="https://CRAN.R-project.org/package=purrr" class="uri">https://CRAN.R-project.org/package=purrr</a>.</p>
</div>
<div id="ref-Jacomy2014-um">
<p>Jacomy, Mathieu, Tommaso Venturini, Sebastien Heymann, and Mathieu Bastian. 2014. “ForceAtlas2, a continuous graph layout algorithm for handy network visualization designed for the Gephi software.” <em>PloS One</em> 9 (6): e98679. doi:<a href="https://doi.org/10.1371/journal.pone.0098679">10.1371/journal.pone.0098679</a>.</p>
</div>
<div id="ref-La_Manno2018-tv">
<p>La Manno, Gioele, Ruslan Soldatov, Amit Zeisel, Emelie Braun, Hannah Hochgerner, Viktor Petukhov, Katja Lidschreiber, et al. 2018. “RNA velocity of single cells.” <em>Nature</em>, August. doi:<a href="https://doi.org/10.1038/s41586-018-0414-6">10.1038/s41586-018-0414-6</a>.</p>
</div>
<div id="ref-Lex2014-hy">
<p>Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and Hanspeter Pfister. 2014. “UpSet: Visualization of Intersecting Sets.” <em>IEEE Transactions on Visualization and Computer Graphics</em> 20 (12): 1983–92. doi:<a href="https://doi.org/10.1109/TVCG.2014.2346248">10.1109/TVCG.2014.2346248</a>.</p>
</div>
<div id="ref-Lun2016-mq">
<p>Lun, Aaron T L, Karsten Bach, and John C Marioni. 2016. “Pooling across cells to normalize single-cell RNA sequencing data with many zero counts.” <em>Genome Biology</em> 17 (1): 1–14. doi:<a href="https://doi.org/10.1186/s13059-016-0947-7">10.1186/s13059-016-0947-7</a>.</p>
</div>
<div id="ref-Lun2019-wp">
<p>Lun, Aaron T L, Samantha Riesenfeld, Tallulah Andrews, The Phuong Dao, Tomas Gomes, participants in the 1st Human Cell Atlas Jamboree, and John C Marioni. 2019. “EmptyDrops: distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data.” <em>Genome Biology</em> 20 (1): 63. doi:<a href="https://doi.org/10.1186/s13059-019-1662-y">10.1186/s13059-019-1662-y</a>.</p>
</div>
<div id="ref-R-SingleCellExperiment">
<p>Lun, Aaron, and Davide Risso. 2019. <em>SingleCellExperiment: S4 Classes for Single Cell Data</em>.</p>
</div>
<div id="ref-McCarthy2017-ql">
<p>McCarthy, Davis J, Kieran R Campbell, Aaron T L Lun, and Quin F Wills. 2017. “Scater: pre-processing, quality control, normalization and visualization of single-cell RNA-seq data in R.” <em>Bioinformatics</em> 33 (8): 1179–86. doi:<a href="https://doi.org/10.1093/bioinformatics/btw777">10.1093/bioinformatics/btw777</a>.</p>
</div>
<div id="ref-McCarthy2012-gc">
<p>McCarthy, Davis J, Yunshun Chen, and Gordon K Smyth. 2012. “Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation.” <em>Nucleic Acids Research</em> 40 (10): 4288–97. doi:<a href="https://doi.org/10.1093/nar/gks042">10.1093/nar/gks042</a>.</p>
</div>
<div id="ref-R-LoomExperiment">
<p>Morgan, Martin, and Daniel Van Twisk. 2019. <em>LoomExperiment: LoomExperiment Container</em>.</p>
</div>
<div id="ref-Patro2017-ka">
<p>Patro, Rob, Geet Duggal, Michael I Love, Rafael A Irizarry, and Carl Kingsford. 2017. “Salmon provides fast and bias-aware quantification of transcript expression.” <em>Nature Methods</em> 14 (4): 417–19. doi:<a href="https://doi.org/10.1038/nmeth.4197">10.1038/nmeth.4197</a>.</p>
</div>
<div id="ref-R-ggforce">
<p>Pedersen, Thomas Lin. 2018. <em>Ggforce: Accelerating ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=ggforce" class="uri">https://CRAN.R-project.org/package=ggforce</a>.</p>
</div>
<div id="ref-R-base">
<p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-Robinson2010-pt">
<p>Robinson, Mark D, Davis J McCarthy, and Gordon K Smyth. 2010. “edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.” <em>Bioinformatics</em> 26 (1): 139–40. doi:<a href="https://doi.org/10.1093/bioinformatics/btp616">10.1093/bioinformatics/btp616</a>.</p>
</div>
<div id="ref-Satija2015-or">
<p>Satija, Rahul, Jeffrey A Farrell, David Gennert, Alexander F Schier, and Aviv Regev. 2015. “Spatial reconstruction of single-cell gene expression data.” <em>Nature Biotechnology</em> 33 (5). Nature Publishing Group: 495–502. doi:<a href="https://doi.org/10.1038/nbt.3192">10.1038/nbt.3192</a>.</p>
</div>
<div id="ref-Scialdone2015-rp">
<p>Scialdone, Antonio, Kedar N Natarajan, Luis R Saraiva, Valentina Proserpio, Sarah A Teichmann, Oliver Stegle, John C Marioni, and Florian Buettner. 2015. “Computational assignment of cell-cycle stage from single-cell transcriptome data.” <em>Methods</em> 85 (September): 54–61. doi:<a href="https://doi.org/10.1016/j.ymeth.2015.06.021">10.1016/j.ymeth.2015.06.021</a>.</p>
</div>
<div id="ref-Sidiropoulos2018-ll">
<p>Sidiropoulos, Nikos, Sina Hadi Sohi, Thomas Lin Pedersen, Bo Torben Porse, Ole Winther, Nicolas Rapin, and Frederik Otzen Bagger. 2018. “SinaPlot: An Enhanced Chart for Simple and Truthful Representation of Single Observations Over Multiple Classes.” <em>Journal of Computational and Graphical Statistics: A Joint Publication of American Statistical Association, Institute of Mathematical Statistics, Interface Foundation of North America</em> 27 (3). Taylor &amp; Francis: 673–76. doi:<a href="https://doi.org/10.1080/10618600.2017.1366914">10.1080/10618600.2017.1366914</a>.</p>
</div>
<div id="ref-Smedley2015-an">
<p>Smedley, Damian, Syed Haider, Steffen Durinck, Luca Pandini, Paolo Provero, James Allen, Olivier Arnaiz, et al. 2015. “The BioMart community portal: an innovative alternative to large, centralized data repositories.” <em>Nucleic Acids Research</em> 43 (W1): W589–98. doi:<a href="https://doi.org/10.1093/nar/gkv350">10.1093/nar/gkv350</a>.</p>
</div>
<div id="ref-Srivastava2019-ce">
<p>Srivastava, Avi, Laraib Malik, Tom Smith, Ian Sudbery, and Rob Patro. 2019. “Alevin efficiently estimates accurate gene abundances from dscRNA-seq data.” <em>Genome Biology</em> 20 (1): 65. doi:<a href="https://doi.org/10.1186/s13059-019-1670-y">10.1186/s13059-019-1670-y</a>.</p>
</div>
<div id="ref-Wickham2010-zq">
<p>Wickham, Hadley. 2010. <em>ggplot2: Elegant Graphics for Data Analysis</em>. Springer New York.</p>
</div>
<div id="ref-R-tidyverse">
<p>———. 2017. <em>Tidyverse: Easily Install and Load the ’Tidyverse’</em>. <a href="https://CRAN.R-project.org/package=tidyverse" class="uri">https://CRAN.R-project.org/package=tidyverse</a>.</p>
</div>
<div id="ref-R-tidyr">
<p>Wickham, Hadley, and Lionel Henry. 2018. <em>Tidyr: Easily Tidy Data with ’Spread()’ and ’Gather()’ Functions</em>. <a href="https://CRAN.R-project.org/package=tidyr" class="uri">https://CRAN.R-project.org/package=tidyr</a>.</p>
</div>
<div id="ref-R-dplyr">
<p>Wickham, Hadley, Romain François, Lionel Henry, and Kirill Müller. 2018. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>.</p>
</div>
<div id="ref-R-cowplot">
<p>Wilke, Claus O. 2019. <em>Cowplot: Streamlined Plot Theme and Plot Annotations for ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=cowplot" class="uri">https://CRAN.R-project.org/package=cowplot</a>.</p>
</div>
<div id="ref-Wolf2018-na">
<p>Wolf, F Alexander, Philipp Angerer, and Fabian J Theis. 2018. “SCANPY: large-scale single-cell gene expression data analysis.” <em>Genome Biology</em> 19 (1): 15. doi:<a href="https://doi.org/10.1186/s13059-017-1382-0">10.1186/s13059-017-1382-0</a>.</p>
</div>
<div id="ref-Wolf2019-fo">
<p>Wolf, F Alexander, Fiona K Hamey, Mireya Plass, Jordi Solana, Joakim S Dahlin, Berthold Göttgens, Nikolaus Rajewsky, Lukas Simon, and Fabian J Theis. 2019. “PAGA: graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells.” <em>Genome Biology</em> 20 (1): 59. doi:<a href="https://doi.org/10.1186/s13059-019-1663-x">10.1186/s13059-019-1663-x</a>.</p>
</div>
<div id="ref-Xie2014-ha">
<p>Xie, Yihui. 2014. “knitr: A Comprehensive Tool for Reproducible Research in R.” In <em>Implementing Reproducible Research</em>, edited by Victoria Stodden, Friedrich Leisch, and Roger D Peng. CRC Press. <a href="https://market.android.com/details?id=book-JcmSAwAAQBAJ" class="uri">https://market.android.com/details?id=book-JcmSAwAAQBAJ</a>.</p>
</div>
<div id="ref-Xie2016-ct">
<p>———. 2016. <em>Dynamic Documents with R and knitr</em>. CRC Press. doi:<a href="https://doi.org/10.1201/b15166">10.1201/b15166</a>.</p>
</div>
<div id="ref-R-knitr">
<p>———. 2018. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in R</em>. <a href="https://CRAN.R-project.org/package=knitr" class="uri">https://CRAN.R-project.org/package=knitr</a>.</p>
</div>
<div id="ref-Xie2018-tw">
<p>Xie, Yihui, J J Allaire, and Garrett Grolemund. 2018. <em>R Markdown: The Definitive Guide</em>. CRC Press LLC. <a href="https://market.android.com/details?id=book-j9YyuAEACAAJ" class="uri">https://market.android.com/details?id=book-j9YyuAEACAAJ</a>.</p>
</div>
<div id="ref-Zappia2018-lz">
<p>Zappia, Luke, and Alicia Oshlack. 2018. “Clustering trees: a visualization for evaluating clusterings at multiple resolutions.” <em>GigaScience</em> 7 (7). doi:<a href="https://doi.org/10.1093/gigascience/giy083">10.1093/gigascience/giy083</a>.</p>
</div>
<div id="ref-Zheng2017-mm">
<p>Zheng, Grace X Y, Jessica M Terry, Phillip Belgrader, Paul Ryvkin, Zachary W Bent, Ryan Wilson, Solongo B Ziraldo, et al. 2017. “Massively parallel digital transcriptional profiling of single cells.” <em>Nature Communications</em> 8 (January): 14049. doi:<a href="https://doi.org/10.1038/ncomms14049">10.1038/ncomms14049</a>.</p>
</div>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<div id="output-files" class="section level2">
<h2>Output files`</h2>
<p>This table describes the output files produced by this document. Right click and <em>Save Link As…</em> to download the results.</p>
<pre class="r"><code>params &lt;- jsonlite::toJSON(params, pretty = TRUE)
readr::write_lines(params,
                   here::here(&quot;output&quot;, DOCNAME, &quot;analysis-parameters.json&quot;))

versions &lt;- purrr::map(versions, as.character)
versions &lt;- jsonlite::toJSON(versions, pretty = TRUE)
readr::write_lines(versions,
                   here::here(&quot;output&quot;, DOCNAME, &quot;package-versions.json&quot;))</code></pre>
<pre class="r"><code>dir.create(here::here(&quot;output&quot;, DOCNAME), showWarnings = FALSE)

knitr::kable(data.frame(
    File = c(
        
    ),
    Description = c(

    )
))</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.0 (2018-04-23)
 os       CentOS release 6.7 (Final)  
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-04-03                  

─ Packages ──────────────────────────────────────────────────────────────
 ! package     * version date       lib source        
   assertthat    0.2.0   2017-04-11 [1] CRAN (R 3.5.0)
   backports     1.1.3   2018-12-14 [1] CRAN (R 3.5.0)
   bindr         0.1.1   2018-03-13 [1] CRAN (R 3.5.0)
   bindrcpp      0.2.2   2018-03-29 [1] CRAN (R 3.5.0)
   broom         0.5.1   2018-12-05 [1] CRAN (R 3.5.0)
   callr         3.1.1   2018-12-21 [1] CRAN (R 3.5.0)
   cellranger    1.1.0   2016-07-27 [1] CRAN (R 3.5.0)
   cli           1.0.1   2018-09-25 [1] CRAN (R 3.5.0)
   colorspace    1.4-0   2019-01-13 [1] CRAN (R 3.5.0)
   crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.0)
   desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.0)
   devtools      2.0.1   2018-10-26 [1] CRAN (R 3.5.0)
   digest        0.6.18  2018-10-10 [1] CRAN (R 3.5.0)
   dplyr       * 0.7.8   2018-11-10 [1] CRAN (R 3.5.0)
   evaluate      0.12    2018-10-09 [1] CRAN (R 3.5.0)
   forcats     * 0.3.0   2018-02-19 [1] CRAN (R 3.5.0)
   fs            1.2.6   2018-08-23 [1] CRAN (R 3.5.0)
   generics      0.0.2   2018-11-29 [1] CRAN (R 3.5.0)
   ggplot2     * 3.1.0   2018-10-25 [1] CRAN (R 3.5.0)
   git2r         0.24.0  2019-01-07 [1] CRAN (R 3.5.0)
   glue        * 1.3.0   2018-07-17 [1] CRAN (R 3.5.0)
   gtable        0.2.0   2016-02-26 [1] CRAN (R 3.5.0)
   haven         2.0.0   2018-11-22 [1] CRAN (R 3.5.0)
   here          0.1     2017-05-28 [1] CRAN (R 3.5.0)
   hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.0)
   htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)
   httr          1.4.0   2018-12-11 [1] CRAN (R 3.5.0)
   jsonlite    * 1.6     2018-12-07 [1] CRAN (R 3.5.0)
   knitr       * 1.21    2018-12-10 [1] CRAN (R 3.5.0)
 P lattice       0.20-35 2017-03-25 [5] CRAN (R 3.5.0)
   lazyeval      0.2.1   2017-10-29 [1] CRAN (R 3.5.0)
   lubridate     1.7.4   2018-04-11 [1] CRAN (R 3.5.0)
   magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.0)
   memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.0)
   modelr        0.1.3   2019-02-05 [1] CRAN (R 3.5.0)
   munsell       0.5.0   2018-06-12 [1] CRAN (R 3.5.0)
 P nlme          3.1-137 2018-04-07 [5] CRAN (R 3.5.0)
   pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.0)
   pkgbuild      1.0.2   2018-10-16 [1] CRAN (R 3.5.0)
   pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.0)
   pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.0)
   plyr          1.8.4   2016-06-08 [1] CRAN (R 3.5.0)
   prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.0)
   processx      3.2.1   2018-12-05 [1] CRAN (R 3.5.0)
   ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.0)
   purrr       * 0.3.0   2019-01-27 [1] CRAN (R 3.5.0)
   R.methodsS3   1.7.1   2016-02-16 [1] CRAN (R 3.5.0)
   R.oo          1.22.0  2018-04-22 [1] CRAN (R 3.5.0)
   R.utils       2.7.0   2018-08-27 [1] CRAN (R 3.5.0)
   R6            2.3.0   2018-10-04 [1] CRAN (R 3.5.0)
   Rcpp          1.0.0   2018-11-07 [1] CRAN (R 3.5.0)
   readr       * 1.3.1   2018-12-21 [1] CRAN (R 3.5.0)
   readxl        1.2.0   2018-12-19 [1] CRAN (R 3.5.0)
   remotes       2.0.2   2018-10-30 [1] CRAN (R 3.5.0)
   rlang         0.3.1   2019-01-08 [1] CRAN (R 3.5.0)
   rmarkdown     1.11    2018-12-08 [1] CRAN (R 3.5.0)
   rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.0)
   rstudioapi    0.9.0   2019-01-09 [1] CRAN (R 3.5.0)
   rvest         0.3.2   2016-06-17 [1] CRAN (R 3.5.0)
   scales        1.0.0   2018-08-09 [1] CRAN (R 3.5.0)
   sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.0)
   stringi       1.2.4   2018-07-20 [1] CRAN (R 3.5.0)
   stringr     * 1.3.1   2018-05-10 [1] CRAN (R 3.5.0)
   testthat      2.0.1   2018-10-13 [4] CRAN (R 3.5.0)
   tibble      * 2.0.1   2019-01-12 [1] CRAN (R 3.5.0)
   tidyr       * 0.8.2   2018-10-28 [1] CRAN (R 3.5.0)
   tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.0)
   tidyverse   * 1.2.1   2017-11-14 [1] CRAN (R 3.5.0)
   usethis       1.4.0   2018-08-14 [1] CRAN (R 3.5.0)
   whisker       0.3-2   2013-04-28 [1] CRAN (R 3.5.0)
   withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.0)
   workflowr     1.1.1   2018-07-06 [1] CRAN (R 3.5.0)
   xfun          0.4     2018-10-23 [1] CRAN (R 3.5.0)
   xml2          1.2.0   2018-01-24 [1] CRAN (R 3.5.0)
   yaml          2.2.0   2018-07-25 [1] CRAN (R 3.5.0)

[1] /group/bioi1/luke/analysis/phd-thesis-analysis/packrat/lib/x86_64-pc-linux-gnu/3.5.0
[2] /group/bioi1/luke/analysis/phd-thesis-analysis/packrat/lib-ext/x86_64-pc-linux-gnu/3.5.0
[3] /group/bioi1/luke/analysis/phd-thesis-analysis/packrat/lib-R/x86_64-pc-linux-gnu/3.5.0
[4] /home/luke.zappia/R/x86_64-pc-linux-gnu-library/3.5
[5] /usr/local/installed/R/3.5.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
