---
title: "Marker genes"
---

```{r knitr, include = FALSE}
DOCNAME = "04-marker-genes"
NOW <- Sys.time()

# Time chunks during knitting
knitr::knit_hooks$set(timeit = function(before) {
    if (before) {
        print(paste("Start:", Sys.time()))
        NOW <<- Sys.time()
    } else {
        print(paste("Stop:", Sys.time()))
        print(Sys.time() - NOW)
    }
})

knitr::opts_chunk$set(
    autodep        = TRUE,
    cache          = TRUE,
    cache.path     = paste0("cache/", DOCNAME, "/"),
    cache.comments = FALSE,
    echo           = TRUE,
    error          = FALSE,
    fig.align      = "center",
    fig.width      = 10,
    fig.height     = 8,
    message        = FALSE,
    warning        = FALSE,
    timeit         = TRUE
)
```

```{r libaries, cache = FALSE}
# scRNA-seq
library("SingleCellExperiment")
library("scater")

# Differential expression
library("edgeR")

# Plotting
library("cowplot")

# Presentation
library("knitr")

# Tidyverse
library("tidyverse")
```

```{r source, cache = FALSE}
source(here::here("R/plotting.R"))
source(here::here("R/output.R"))
```

```{r depends-paths}
clust_path <- here::here("data/processed/03-clustered.Rds")
```

```{r bpparam, cache = FALSE}
bpparam5 <- BiocParallel::MulticoreParam(workers = 5)
bpparam3 <- BiocParallel::MulticoreParam(workers = 3)
```

Introduction
============

In this document we are going to identify marker genes for the clusters
previously defined using `Seurat` using the `edgeR` package and use these genes
to assign cell type labels to the clusters.

```{r load, cache.extra = tools::md5sum(clust_path)}
if (file.exists(clust_path)) {
    sce <- read_rds(clust_path)
} else {
    stop("Clustered dataset is missing. ",
         "Please run '03-clustering.Rmd' first.",
         call. = FALSE)
}

n_clusts <- length(unique(colData(sce)$Cluster))
```

Filtering
=========

Before testing we will perform some additional gene filtering. We are trying to
identify markers that are highly expressed in a cluster compared to other cells
so we can remove genes that are not sufficiently expressed in any cluster (have
a low maximum mean cluster expression).

```{r filter}
cluster_means <- map(levels(colData(sce)$Cluster), function(clust) {
    calcAverage(sce[, colData(sce)$Cluster == clust])
})
cluster_means <- do.call("cbind", cluster_means)
rowData(sce)$ClusterMaxMean <- rowMax(cluster_means)
rowData(sce)$log10_ClusterMaxMean <- log10(rowData(sce)$ClusterMaxMean)

outlierHistogram(as.data.frame(rowData(sce)), "log10_ClusterMaxMean",
                 mads = seq(0.5, 3, 0.5))

maxmean_mads  <- 0.5
maxmean_out   <- isOutlier(rowData(sce)$log10_ClusterMaxMean,
                           nmads = maxmean_mads, type = "lower")
maxmean_thresh <- attr(maxmean_out, "thresholds")["lower"]

sce_filt <- sce[rowData(sce)$log10_ClusterMaxMean > maxmean_thresh, ]
```

Selecting a threshold of `r maxmean_thresh` removes `r sum(maxmean_out)` genes.

Fitting
=======

We will use `edgeR` for differential expression testing to identify marker
genes for our clusters. Although originally designed for bulk RNA-seq comparisons
have shown that `edgeR` also has good performance for scRNA-seq. We will test
each cluster against all the other cells in the dataset. This may miss subtler 
differences between similar cell types but should identify key markers for each 
cluster and is computationally easier than completing every pairwise comparison.

First we construct a design matrix and `DGEList` object then we calculate
normalisation factors, estimate dispersions and fit the model.

```{r fit, cache.lazy = FALSE}
# Cell detection rate
det_rate <- scale(Matrix::colMeans(counts(sce_filt) > 0))

# Create design matrix
design <- model.matrix(~ 0 + det_rate + colData(sce_filt)$Cluster)
colnames(design) <- c("DetRate", paste0("C", seq_len(n_clusts) - 1))

# Create DGEList
dge <- DGEList(counts(sce_filt))
# Add gene annotation
dge$genes <- rowData(sce_filt)[, c("Name", "ID", "entrezgene", "description")]
# Calculate norm factors
dge <- calcNormFactors(dge)
# Estimate dispersions
dge <- estimateDisp(dge, design)
# Fit model
de_fit <- glmQLFit(dge, design = design)
```

Before we perform the tests let's check how good the fit is.

Mean
----

```{r mean-fit}
plot_data <- tibble(Observed = log1p(Matrix::rowMeans(counts(sce_filt))),
                    Expected = log1p(rowMeans(de_fit$fitted.values))) %>%
    mutate(Mean = 0.5 * (Observed + Expected),
           Difference = Observed - Expected)

ggplot(plot_data, aes(x = Mean, y = Difference)) +
    geom_point() +
    geom_smooth(method = "loess") +
    geom_hline(yintercept = 0, colour = "red") +
    xlab("0.5 * (Observed + Expected)") +
    ylab("Observed - Expected") +
    ggtitle("Fit of gene means") +
    theme_minimal()
```

Zeros
-----

```{r zeros-fit}
plot_data <- tibble(Observed = Matrix::rowMeans(counts(sce_filt) == 0),
                    Expected = rowMeans(
                        (1 + de_fit$fitted.values * dge$tagwise.dispersion) ^ 
                            (-1 / dge$tagwise.dispersion)
                    )) %>%
    mutate(Mean = 0.5 * (Observed + Expected),
           Difference = Observed - Expected)

ggplot(plot_data, aes(x = Mean, y = Difference)) +
    geom_point() +
    geom_smooth(method = "loess") +
    geom_hline(yintercept = 0, colour = "red") +
    xlab("0.5 * (Observed + Expected)") +
    ylab("Observed - Expected") +
    ggtitle("Fit of gene proportion of zeros") +
    theme_minimal()
```

BCV
---

```{r bcv}
plotBCV(dge)
```

QL Dispersions
--------------

```{r qldisp}
plotQLDisp(de_fit)
```

These plots suggest that the fit is fairly good and there isn't an excessive
amount of zero inflation we would need to account for using other methods.

Testing
=======

Once we have a fitted model we can test each of our contrasts.

```{r test, cache.lazy = FALSE}
de_tests <- bplapply(seq_len(n_clusts), function(idx) {
    contrast <- rep(-1 / (n_clusts - 1), n_clusts)
    contrast[idx] <- 1
    contrast <- c(0, contrast)
    glmQLFTest(de_fit, contrast = contrast)
}, BPPARAM = bpparam5)
```

```{r summarise-tests}
de_list <- lapply(seq_len(n_clusts), function(idx) {
    topTags(de_tests[[idx]], n = Inf, sort.by = "logFC")$table %>%
        as.data.frame() %>%
        mutate(Cluster = idx - 1) %>%
        select(Cluster, everything())
})
names(de_list) <- paste("Cluster", 0:(n_clusts - 1))

fdr_thresh <- 0.05
logFC_thresh <- 1

markers_list <- lapply(de_list, function(de) {
    de %>%
        filter(FDR < fdr_thresh, logFC > logFC_thresh) %>%
        select(Name, logFC, PValue, FDR)
})
```

Before we look at the gene lists let's check some diagnostic plots.

P-values {.tabset}
--------

Distribution of p-values for each cluster should be uniform with a peak near
zero.

```{r pvals, results = "hide"}
src_list <- lapply(seq_len(n_clusts) - 1, function(clust) {
    src <- c(
        "### Cluster {{clust}} {.unnumbered}",
        "```{r pvals-{{clust}}}",
        "ggplot(de_list[[{{clust}} + 1]], aes(x = PValue)) +",
        "geom_histogram() +",
        "theme_minimal()",  
        "```",
        ""
    )
    knit_expand(text = src)
})

out <- knit_child(text = unlist(src_list), options = list(cache = FALSE))
```

`r out`

Mean-difference {.tabset}
---------------

Mean-difference plot highlighting significant genes.

```{r md, results = "hide"}
src_list <- lapply(seq_len(n_clusts) - 1, function(clust) {
    src <- c(
        "### Cluster {{clust}} {.unnumbered}",
        "```{r md-{{clust}}}",
        "plotMD(de_tests[[{{clust}} + 1]])",  
        "```",
        ""
    )
    knit_expand(text = src)
})

out <- knit_child(text = unlist(src_list), options = list(cache = FALSE))
```

`r out`

Summary
-------

```{r de-summary}
plot_data <- de_list %>%
    bind_rows() %>%
    filter(FDR < 0.05) %>%
    mutate(IsUp = logFC > 0) %>%
    group_by(Cluster) %>%
    summarise(Up = sum(IsUp), Down = sum(!IsUp)) %>%
    mutate(Down = -Down) %>%
    gather(key = "Direction", value = "Count", -Cluster) %>%
    mutate(Cluster = factor(Cluster, levels = 0:(n_clusts - 1)))

ggplot(plot_data,
       aes(x = fct_rev(Cluster), y = Count, fill = Direction)) +
    geom_col() +
    geom_text(aes(y = Count + sign(Count) * max(abs(Count)) * 0.07,
                  label = abs(Count)),
              size = 6, colour = "grey25") +
    coord_flip() +
    scale_fill_manual(values = c("#377eb8", "#e41a1c")) +
    ggtitle("Number of identified genes") +
    theme_minimal() +
    theme(axis.title = element_blank(),
          axis.line = element_blank(),
          axis.ticks = element_blank(),
          axis.text.x = element_blank(),
          legend.position = "bottom")
```

Table {.tabset}
-----

Table of significant genes for each cluster with log fold-change greater than
one.

```{r table, results = "hide"}
src_list <- lapply(seq_len(n_clusts) - 1, function(clust) {
    src <- c(
        "### Cluster {{clust}} {.unnumbered}",
        "```{r table-{{clust}}}",
        "markers_list[[{{clust}} + 1]]",
        "```",
        ""
    )
    knit_expand(text = src)
})

out <- knit_child(text = unlist(src_list), options = list(cache = FALSE))
```

`r out`

Known genes {.tabset}
===========

To help with identifying clusters we will look at some previously identified
marker genes.

```{r known-genes}
known_genes <- c(
    # Stroma
    "TAGLN", "ACTA2", "MAB21L2", "DLK1", "GATA3", "COL2A1", "COL9A3",
    # Podocyte
    "PODXL", "NPHS2", "TCF21",
    # Cell cycle
    "HIST1H4C", "PCLAF", "CENPF", "HMGB2",
    # Endothelium
    "CLDN5", "PECAM1", "KDR", "CALM1",
    # Neural
    "TTYH1", "SOX2", "HES6", "STMN2",
    # Epithelium
    "PAX2", "PAX8", "KRT19",
    # Muscle
    "MYOG", "MYOD1",
    # Immune
    "TYROBP", "MPO", "S100A9"
)

known_types <- factor(c(
    rep("Stroma", 7),
    rep("Podocyte", 3),
    rep("Cell cycle", 4),
    rep("Endothelium", 4),
    rep("Neural", 4),
    rep("Epithelium", 3),
    rep("Muscle", 2),
    rep("Immune", 3)
))
names(known_types) <- known_genes
```

Expression
----------

Size indicates (scaled) mean expression level, transparency indicate proportion
of cells expressing the gene.

```{r known-genes-expr}
plot_data <- map_df(seq_len(n_clusts), function(idx) {
    clust_cells <- colData(sce_filt)$Cluster == idx - 1
    means <- Matrix::rowMeans(logcounts(sce_filt)[known_genes, clust_cells])
    props <- Matrix::rowSums(counts(sce_filt)[known_genes, clust_cells] > 0) /
                                 sum(clust_cells)
    tibble(Cluster = idx - 1,
           Gene = known_genes,
           Type = known_types,
           Mean = means,
           Prop = props)
}) %>%
    mutate(Cluster = factor(Cluster, levels = (n_clusts - 1):0),
           Gene = factor(Gene, levels = known_genes),
           Type = factor(Type, levels = unique(known_types))) %>%
    group_by(Gene) %>%
    mutate(Mean = scale(Mean))

ggplot(plot_data,
       aes(x = Gene, y = Cluster, size = Prop, alpha = Mean, colour = Type)) +
    geom_point() +
    guides(size = FALSE, alpha = FALSE) +
    facet_grid(~ Type, scales = "free_x", space = "free_x") +
    theme_minimal() +
    theme(axis.title.x = element_blank(),
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none")
```

Significance
------------

Size indicates log fold-change and transparency indicates significance. As we
are only interested in marker genes negative fold-changes have been set to zero
and significance for those cases has been set to one.

```{r known-genes-sig}
plot_data <- map_df(seq_len(n_clusts), function(idx) {
    de_list[[idx]] %>%
        filter(Name %in% known_genes)
}) %>%
    mutate(FDR = if_else(logFC < 0, 1, FDR)) %>%
    mutate(logFC = if_else(logFC < 0, 0, logFC)) %>%
    mutate(Type = known_types[Name]) %>%
    mutate(Cluster = factor(Cluster, levels = (n_clusts - 1):0),
           Gene = factor(Name, levels = known_genes),
           Type = factor(Type, levels = unique(known_types)))

ggplot(plot_data,
       aes(x = Gene, y = Cluster, size = logFC, alpha = -FDR, colour = Type)) +
    geom_point() +
    facet_grid(~ Type, scales = "free_x", space = "free_x") +
    theme_minimal() +
    theme(axis.title.x = element_blank(),
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none")
```

Figures
=======

```{r figure}
plot_data <- tibble(Observed = log1p(Matrix::rowMeans(counts(sce_filt))),
                    Expected = log1p(rowMeans(de_fit$fitted.values))) %>%
    mutate(Mean = 0.5 * (Observed + Expected),
           Difference = Observed - Expected)

mean_plot <- ggplot(plot_data, aes(x = Mean, y = Difference)) +
    geom_point(alpha = 0.3, colour = "#7A52C7") +
    geom_smooth(method = "loess", colour = "#00ADEF",
                fill = "#00ADEF", size = 1) +
    geom_hline(yintercept = 0, colour = "#EC008C", size = 1) +
    xlab("0.5 * (Observed mean + Expected mean)") +
    ylab("Observed mean - Expected mean") +
    ggtitle("Fit of gene means") +
    theme_minimal()

plot_data <- tibble(Observed = Matrix::rowMeans(counts(sce_filt) == 0),
                    Expected = rowMeans(
                        (1 + de_fit$fitted.values * dge$tagwise.dispersion) ^ 
                            (-1 / dge$tagwise.dispersion)
                    )) %>%
    mutate(Mean = 0.5 * (Observed + Expected),
           Difference = Observed - Expected)

zeros_plot <- ggplot(plot_data, aes(x = Mean, y = Difference)) +
    geom_point(alpha = 0.3, colour = "#7A52C7") +
    geom_smooth(method = "loess", colour = "#00ADEF",
                fill = "#00ADEF", size = 1) +
    geom_hline(yintercept = 0, colour = "#EC008C", size = 1) +
    xlab("0.5 * (Observed proportion zeros + Expected proportion zeros)") +
    ylab("Observed proportion zeros - Expected proportion zeros") +
    ggtitle("Fit of gene proportion of zeros") +
    theme_minimal()

plot_data <- tibble(
    AvgLogCPM = dge$AveLogCPM,
    Tagwise = dge$tagwise.dispersion,
    Trended = dge$trended.dispersion
)

bcv_plot <- ggplot(plot_data, aes(x = AvgLogCPM)) +
    geom_point(aes(y = sqrt(Tagwise)), alpha = 0.3, colour = "#7A52C7") +
    geom_hline(yintercept = sqrt(dge$common.dispersion),
               colour = "#EC008C", size = 1) +
    geom_line(aes(y = sqrt(Trended)), colour = "#00ADEF", size = 1) +
    ggtitle("edgeR biological coefficient of variation") +
    xlab(expression("Average counts per million ("*log["2"]*")")) +
    ylab("Biological coefficient of variation") +
    theme_minimal()

plot_data <- de_list %>%
    bind_rows() %>%
    filter(FDR < 0.05) %>%
    mutate(IsUp = logFC > 0) %>%
    filter(IsUp) %>%
    mutate(MaxLogFC = max(logFC)) %>%
    group_by(Cluster) %>%
    mutate(logFCScaled = scales::rescale(
        logFC, from = c(0, max(MaxLogFC)), to = c(0, n()))
    ) %>%
    ungroup() %>%
    mutate(Cluster = factor(Cluster, levels = 0:(n_clusts - 1)))

counts_plot <- ggplot(plot_data, aes(x = Cluster)) +
    geom_bar(aes(fill = Cluster), alpha = 0.3) +
    geom_jitter(aes(y = logFCScaled, colour = logFC),
                position = position_jitter(height = 0, width = 0.35, seed = 1),
                alpha = 0.3) +
    geom_text(aes(label=..count..), stat = "count",
              vjust = -0.5, size = 4, colour = "grey10") +
    scale_fill_discrete(guide = FALSE) +
    scale_colour_viridis_c(option = "plasma") +
    ggtitle("Up-regulated DE genes by cluster") +
    theme_minimal() +
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.1, 
                                    vjust = 1, margin = margin(1)),
          axis.title = element_blank(),
          axis.text.y = element_blank(),
          panel.grid = element_blank())

plot_data <- map_df(seq_len(n_clusts), function(idx) {
    de_list[[idx]] %>%
        filter(Name %in% known_genes)
}) %>%
    mutate(FDR = if_else(logFC < 0, 1, FDR)) %>%
    mutate(logFC = if_else(logFC < 0, 0, logFC)) %>%
    mutate(Type = known_types[Name]) %>%
    mutate(Cluster = factor(Cluster, levels = (n_clusts - 1):0),
           Gene = factor(Name, levels = known_genes),
           Type = factor(Type, levels = unique(known_types)))

genes_plot <- ggplot(plot_data,
       aes(x = Gene, y = Cluster, size = logFC, alpha = -FDR, colour = Type)) +
    geom_point() +
    facet_grid(~ Type, scales = "free_x", space = "free_x") +
    ggtitle("edgeR results for some published markers by cluster") +
    theme_minimal() +
    theme(axis.title = element_blank(),
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none")

p1 <- plot_grid(mean_plot, zeros_plot, bcv_plot, counts_plot,
                ncol = 2, labels = "AUTO")

fig <- plot_grid(p1, genes_plot, ncol = 1, labels = c("", "E"),
                 rel_heights = c(1, 0.5))

ggsave(here::here("output", DOCNAME, "de-results.pdf"), fig,
       width = 7, height = 10, scale = 1.5)
ggsave(here::here("output", DOCNAME, "de-results.png"), fig,
       width = 7, height = 10, scale = 1.5)

fig
```

Summary
=======

Based on the detected markers we will assign the following cell types to the
clusters.

```{r assignment}
assignments <- tribble(
    ~Cluster,         ~Assignment,
         "0",            "Stroma",
         "1",            "Stroma",
         "2",        "Cell cycle",
         "3",            "Stroma",
         "4",       "Endothelium",
         "5",          "Podocyte",
         "6",   "Possible immune",
         "7",            "Stroma",
         "8",       "Endothelium",
         "9",             "Glial",
        "10",        "Epithelium",
        "11", "Muscle progenitor",
        "12", "Neural progenitor"
)

assignments
```

Parameters
----------

This table describes parameters used and set in this document.

```{r parameters, cache.lazy = FALSE}
params <- list(
    list(
        Parameter = "maxmean_thresh",
        Value = maxmean_thresh,
        Description = "Minimum threshold for (log10) maxium cluster mean"
    ),
    list(
        Parameter = "maxmean_mads",
        Value = maxmean_mads,
        Description = "MADs for (log10) maxium cluster mean"
    ),
    list(
        Parameter = "fdr_thresh",
        Value = fdr_thresh,
        Description = "FDR threshold for significant marker genes"
    ),
    list(
        Parameter = "logFC_thresh",
        Value = logFC_thresh,
        Description = "Log foldchange threshold for significant marker genes"
    )
)

params <- jsonlite::toJSON(params, pretty = TRUE)
knitr::kable(jsonlite::fromJSON(params))
```

Output files
------------

This table describes the output files produced by this document. Right click
and _Save Link As..._ to download the results.

```{r save}
write_rds(sce, here::here("data/processed/04-markers.Rds"))
write_rds(de_fit, here::here("data/processed/04-DGEGLM.Rds"))
```

```{r output}
dir.create(here::here("output", DOCNAME), showWarnings = FALSE)

readr::write_lines(params, here::here("output", DOCNAME, "parameters.json"))
writeGeneTable(de_list, here::here("output", DOCNAME, "de_genes.xlsx"))
writeGeneTable(bind_rows(de_list),
               here::here("output", DOCNAME, "de_genes.csv"))
readr::write_tsv(assignments, here::here("output", DOCNAME, "assignments.tsv"))

knitr::kable(data.frame(
    File = c(
        getDownloadLink("parameters.json", DOCNAME),
        getDownloadLink("de_genes.xlsx", DOCNAME),
        getDownloadLink("de_genes.csv.zip", DOCNAME),
        getDownloadLink("assignments.csv", DOCNAME),
        getDownloadLink("de-results.png", DOCNAME),
        getDownloadLink("de-results.pdf", DOCNAME)
    ),
    Description = c(
        "Parameters set and used in this analysis",
        paste("Results of marker gene testing in XLSX format with one tab",
              "per cluster"),
        "Results of marker gene testing in zipped CSV format",
        "Cluster assignments (TSV)",
        "DE results figure (PNG)",
        "DE results figure (PDF)"
    )
))
```

Session information
-------------------

```{r session-info, cache = FALSE}
devtools::session_info()
```
